packages/api-server/src/core/cache/CacheBuilder.ts

```diff
@@ -22,23 +22,20 @@ class NoneCacheProvider implements CacheProvider {
 
 export default class CacheBuilder {
 
-    private normalCacheProvider?: CacheProvider;
+    private readonly normalCacheProvider?: CacheProvider;
 
-    private cachedTableCacheProvider?: CacheProvider;
+    private readonly cachedTableCacheProvider?: CacheProvider;
 
     private noneCacheProvider: CacheProvider = new NoneCacheProvider();
 
-    private enableCache = true;
-
-    constructor(readonly log: pino.Logger, enableCache: boolean) {
-        this.enableCache = enableCache;
-        if (enableCache) {
-            // TODO: getConnectionOptions() should be overridable by tests.
-            const normalCacheConn = new ConnectionWrapper(getConnectionOptions());
-            this.normalCacheProvider = new NormalTableCacheProvider(normalCacheConn);
-
-            const cachedTableCacheConn = new ConnectionWrapper(getConnectionOptions());
-            this.cachedTableCacheProvider = new CachedTableCacheProvider(cachedTableCacheConn);
+    constructor(
+        private readonly log: pino.Logger,
+        private readonly enableCache: boolean = false,
```

Comment:
TODO: When would we have a disabled `CacheBuilder` with a connection? The `enableCache` could be a computed filed.

Reply:
Good idea, but, the code for creating instances should also be changed to:

```typescript
new CacheBuilder(log, process.env.ENABLE_CACHE ? conn : undefined)
```

Reply:
I may prefer the current constructor, because we can:

Use the connection provided from the outside:

```typescript
new CacheBuilder(log, process.env.ENABLE_CACHE, conn)
```

Use the default connection created internally:

```typescript
new CacheBuilder(log, process.env.ENABLE_CACHE)
```

Also, we should not associate the two conditions of whether to use the cache and whether there is a database connection together, because its provider can be redis.

Reply:
Why not just provide `CacheProvider` instead of `Connection` as argument
```ts
new CacheBuilder(log, noCacheProvider)
new CacheBuilder(log, tableCacheProvider)
new CacheBuilder(log, redisCacheProvider)
```

Reply:
will fix in #956